src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Frameworks/SdkSupportedTargetPlatformVersionProvider.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.Threading;
+using EnumCollection = System.Collections.Generic.ICollection<Microsoft.VisualStudio.ProjectSystem.Properties.IEnumValue>;
+using EnumCollectionProjectValue = Microsoft.VisualStudio.ProjectSystem.IProjectVersionedValue<System.Collections.Generic.ICollection<Microsoft.VisualStudio.ProjectSystem.Properties.IEnumValue>>;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Frameworks
+{
+    /// <summary>
+    ///     Responsible for producing valid values for the SdkSupportedTargetPlatformVersion property from a design-time build.
+    /// </summary>
+    [ExportDynamicEnumValuesProvider("SdkSupportedTargetPlatformVersionEnumProvider")]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal class SdkSupportedTargetPlatformVersionProvider : ChainedProjectValueDataSourceBase<EnumCollection>, IDynamicEnumValuesProvider, IDynamicEnumValuesGenerator
+    {
+        private readonly IProjectSubscriptionService _subscriptionService;
+
+        [ImportingConstructor]
+        public SdkSupportedTargetPlatformVersionProvider(
+            ConfiguredProject project,
+            IProjectSubscriptionService subscriptionService)
+            : base(project, synchronousDisposal: true, registerDataSource: false)
+        {
+            _subscriptionService = subscriptionService;
+
+            ReadyToBuild = new OrderPrecedenceImportCollection<IConfiguredProjectReadyToBuild>(projectCapabilityCheckProvider: project);
+        }
+
+        [ImportMany]
+        public OrderPrecedenceImportCollection<IConfiguredProjectReadyToBuild> ReadyToBuild
```

Comment:
These are not needed as this is entirely evaluation (not build) based.

Reply:
This is addressed in the new commits, thanks.